docs/vc-http-api.yml

```diff
@@ -592,6 +651,91 @@ components:
           $ref: "#/components/schemas/PresentCredentialOptions"
     ProvePresentationResponse:
       $ref: "#/components/schemas/VerifiablePresentation"
+    NotifyPresentationAvailableRequest:
+      type: object
+      properties:
+        query:
+          type: object
+          description: See https://w3c-ccg.github.io/vp-request-spec/#format
+          properties:
+            type:
+              type: string
+              description: "The type of query the server should reply with."
+            credentialQuery:
+              type: object
+              description: "Details of the client's available presentation"
+
+      example:
+        {
+          "query":
+            [
+              {
+                "type": "RequestQueryByFrame",
+                "credentialQuery":
+                  [
+                    {
+                      "type":
+                        [
+                          "VerifiableCredential",
+                          "CommercialInvoiceCertificate",
+                        ],
+                      "reason": "Wallet XYZ is ready to selectively disclose new credentials.",
+                    },
+                  ],
+              },
+            ],
+        }
+
+    NotifyPresentationAvailableResponse:
+      type: object
+      properties:
+        query:
```

Comment:
optional?
What if a client cannot meet the requirements of the new query, is there a need for a follow-up call? How does the server protect against spam notifications?

Reply:
out of scope, the same issue regarding spam exists for all the other endpoints... and there is no authorization defined... which would appear to be a prerequisite for spam mitigation.

I pulled query from https://w3c-ccg.github.io/vp-request-spec/#query-by-example

which is exactly how CHAPI works,

Imo if you return a 400+ as the server to a notification request, each of those cases would seem to be need to be defined.... including https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/429